### ISSUE TITLE
Usage of tooltip formatter : onClick stops working on data refresh

### ISSUE THREAD

**Issue by Dubes** [whawker/react-jsx-highcharts#77]
**What I am trying to achieve**
I need to show/hide tooltip based on some logic. As per highcharts [documentation](https://api.highcharts.com/highcharts/tooltip.formatter), the correct way is to use a formatter function returning the text or false. For this, I need both: `this.point.x` provided by highcharts and `this.props.data` which is passed to the chart instance.
I also have a need to update the chart data on some user action. 

** Where I am stuck **
The onClick of Series data stops when I use the formatter function. 
I forked the sandbox you had created for issue #70 to reproduce the problem. https://codesandbox.io/s/922kn8wy6r 

I cannot bind the "formatter" function to the Chart instance, as I need the `this.point.x` in the formatter function. 

**What I have tried so far** 

- Creating the function in render() method of the chart --> this interferes with the onClick behavior of series. 
- Passing the function as props from the container.
- Higher Order function to create this function --> I am pretty sure I am doing it wrong.
- Using the Highcharts callback. I can get to the tooltip by: chart.tooltip, but I do not see the "formatter" option anywhere underneath it. 

Any advice or hint would be appreciated. Also open to contributing a fix if required, but would need some handholding.

**Comment by whawker** [comment]
Hi @Dubes, you just need to watch out where you are using arrow functions, as they change the `this` scope to be the outer scope - in React's case, the component's scope.

By using a normal function, it will keep the `this` to the original scope that Highcharts sets, allowing you to use `this.point.x` as expected.

I've updated you code here: https://codesandbox.io/s/k7mr4ovq3

**Comment by Dubes** [comment]
Thanks for the hint about the arrow function. It is the scope where I am stuck, the equality check in the example would eventually come from props, both of which belong in different scopes.
```
 formatter = function() {
    // the 1 will come from "props.toolTipEnabledFor"
    return this.point.x === 1 ? "Yeah" : false;
  `};
```

I tried with a higher order function, but it works only in constructor... not when I call it on react lifecycle or passing the function via props. 
https://codesandbox.io/s/llzzl17jnz

**Comment by whawker** [comment]
Well good news and bad news.

Good news, you can use both props, and the Highcharts tooltip formatter scope together, using your `makeFormatterFunction`, see here: https://codesandbox.io/s/m38pnvvv78

Bad news, seems the events which I dynamically added are removed when a series updates [see here](https://github.com/highcharts/highcharts/blob/master/js/parts/Dynamics.js#L891), so I'll need to add a fix for that.

**Comment by whawker** [comment]
Just published version 2.2.0 which include a fix for this bug.

You can see the example is fixed [here](https://whawker.github.io/react-jsx-highcharts/examples/UpdateWithEvents/index.html) which is using the same code as the previous CodeSandbox.

**Comment by Dubes** [comment]
That was quick!!! thanks a lot!
